### PR TITLE
Optimobile foundations

### PR DESCRIPTION
**Pull request by BBHOptimove**
Add foundations of Optimobile functionality by combining base elements from the Kumulos SDK into the Optimove SDK.

### CODE REVIEW

**Comment by BBHOptimove** [comment]
optimobile pr to dev

File: Optimobile/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//  Created by Barak Ben Hur on 04/04/2022.
+//
+
+import UIKit
+import KumulosSDK
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
```
Comment (optimove-andy) [inline Optimobile/AppDelegate.swift]: Im not sure im following yet - this file looks like the AppDelegate for an app and has storyboards etc? Should this all be deleted or is it a test app shipping with the SDK? From the diff it seems to be new.

File: OptimoveSDK/Sources/Classes/Optimobile.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    
+    public static var sheard: Optimobile {
+        get {
+            return internalQueue.sync {
+                _sheard
+            }
+        }
+    }
+    
+    public typealias deepLinkComplition = (_ deepLinkResolution: DeepLinkResolution) -> ()
```
Comment (optimove-andy) [inline OptimoveSDK/Sources/Classes/Optimobile.swift]: minor spelling note "Complition" -> "Completion"

File: OptimoveSDK/Sources/Classes/Optimobile.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+             pushOpened(responder: pushOpenedComplition?)
+    }
+    
+    private static var builder: KSConfigBuilder!
+    
+    private init() {}
+    
+    public static func configure(for tenantInfo: OptimoveTenantInfo, apiKey: String, secretKey: String, abilities: [Abilities]? = nil) {
+        Optimove.configure(for: tenantInfo)
+        builder = KSConfigBuilder(apiKey: apiKey, secretKey: secretKey)
```
Comment (optimove-andy) [inline OptimoveSDK/Sources/Classes/Optimobile.swift]: Maybe loop Kosta in on this question but i think we should look to hide any Kumulos / KS references, even if we just renamed the class to something like OptiMobile / OptimobileConfigBuilder

File: Optimobile/AppDelegate.swift (hunk 1 above)
Comment (cgwyllie) [reply Optimobile/AppDelegate.swift]: I think these files were used during prototyping, have removed them form the PR.

File: OptimoveSDK/Sources/Classes/Optimobile.swift (hunk 3 above)
Comment (cgwyllie) [reply OptimoveSDK/Sources/Classes/Optimobile.swift]: Renaming passes to follow

File: OptimoveSDK/Sources/Classes/Observers/AppOpenObserver.swift [hunk 4]
```diff
@@ -1,60 +0,0 @@
-//  Copyright © 2019 Optimove. All rights reserved.
-
-import Foundation
-import OptimoveCore
-import UIKit.UIApplication
-
-final class AppOpenObserver {
```
Comment (cgwyllie) [inline OptimoveSDK/Sources/Classes/Observers/AppOpenObserver.swift]: I'm not sure if this behaviour should be removed, is it Optipush-specific?

Comment (cgwyllie) [reply OptimoveSDK/Sources/Classes/Observers/AppOpenObserver.swift]: Restored

File: OptimoveSDK/Sources/Classes/Optimobile.swift (hunk 2 above)
Comment (cgwyllie) [reply OptimoveSDK/Sources/Classes/Optimobile.swift]: Renaming pass to follow when combining the public APIs

**Pull request merged**